weave

```diff
@@ -97,6 +99,32 @@ collect_cidr_args() {
     fi
 }
 
+is_ip() {
+    # The regexp here is far from precise, but good enough.
+    echo "$1" | grep -E '^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$' >/dev/null
```

Comment:
perhaps define an `ip_regexp='[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'` and use that both here and in `is_cidr`.